src/formatter.rs

```diff
@@ -176,6 +176,11 @@ pub struct ColorFormatter<W> {
     cached_styles: HashMap<Vec<String>, Style>,
     /// The style we last wrote to the output.
     current_style: Style,
+    /// Whether the style requested by the current labels still needs to be
+    /// written to the terminal. We don't write the style to the terminal
+    /// until we actually write some text. This is just an optimization; there
+    /// should be no functional difference if this is always `true`.
+    has_pending_style_change: bool,
```

Comment:
I think this flag is redundant since we have `current_style` (actual) and `requested_style()` (expected).

This might serve as an optimization to avoid `cached_styles` lookup, but I don't know if that would matter.

Reply:
You're probably right. I guess it would mostly matter if we printed lots of text without changing style (so this flag would let us skip the check), but we rarely do that, so it probably doesn't matter. I'll drop the flag.